### PR TITLE
Parachain Staking Milestone 2: UI Dashboard

### PR DESCRIPTION
**Pull request by 4meta5**
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] The [invoice form :pencil:](https://forms.gle/LSRr7PCjBpEbKGh89) has been filled out for this milestone.
- [x] This pull request is being made by the same account as the accepted application.
- [x] In case of acceptance, the payment will be transferred to the BTC/ETH payment address provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://github.com/w3f/Grants-Program/blob/master/docs/milestone-deliverables-guidelines.md).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/389

### CODE REVIEW

File: deliveries/parachain_staking_2.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Milestone Delivery :mailbox:
+
+**The [invoice form :pencil:](https://docs.google.com/forms/d/e/1FAIpQLSfmNYaoCgrxyhzgoKQ0ynQvnNRoTmgApz9NrMp-hd8mhIiO0A/viewform) has been filled out correctly for this milestone and the delivery is according to the official [milestone delivery guidelines](https://github.com/w3f/General-Grants-Program/blob/master/grants/milestone-deliverables-guidelines.md).**  
+
+* **PR Link:** https://github.com/w3f/Grants-Program/pull/389
```
Comment (4meta5) [inline deliveries/parachain_staking_2.md:5]:
```suggestion
* **Application Document:** https://github.com/w3f/Grants-Program/blob/master/applications/parachain-staking.md
```

**Comment by semuelle** [comment]
Thank you for the delivery, Amar. We will look into it as soon as possible.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks again for the delivery. I quickly looked into it and everything seems to work. However, the UI only provides an overview of the current staking initiatives and doesn’t actually let you participate in the staking. Is this on-purpose? I know the deliveries were unspecific in the contract, but I would have expected that you can also stake via the UI.

**Comment by 4meta5** [comment]
> Doesn't actually let you participate in the staking. Is this on-purpose? I know the deliveries were unspecific in the contract, but I would have expected that you can also stake via the UI.

OK, we will add this functionality, but it will require another 2 weeks. I'll let you know as soon as it is done.

Our perspective was that the deliverable needed to display meaningful statistics to educate user staking through whatever custom UI is provided by the implementation. For moonbeam/moonriver, we prefer users use the staking dapp provided by PureStake (https://apps.moonbeam.network/moonriver).

**Comment by Noc2** [comment]
Thanks for the quick reply. I think it would be beneficial if you add it directly to the UI. Alternatively you could also link to your own staking dapp, to at least inform users about how they can participate.

**Comment by alxs** [comment]
@4meta5 any updates?

**Comment by 4meta5** [comment]
@alxs We decided to add the `delegate` feature to the UI dashboard but it is currently in progress and will be done soon. Our best TS dev @joelamouche is working on it. He was busy with tasks for Moonbeam launch, but is back on this now.

**Comment by alxs** [comment]
@4meta5 @joelamouche checking in again. Could you provide an ETA?

**Comment by crystalin** [comment]
@alxs Sorry, @joelamouche is not working on this project anymore. 
Our team took over and I believe it is close to completion. Let me come back you today

**Comment by purestaketdb** [comment]
I am onboarding @fxgamundi to the project this week, he'll need to get up to speed on the Polkadot.js libraries but is the developer on our own staking dapp. I'll have more clarity about timelines next week. Note, basic staking is working but the pallet was extended in December and those changes need to be rolled in.

**Comment by alxs** [comment]
Thanks @crystalin @purestaketdb, sounds good. Please keep us posted.

**Comment by fxgamundi** [comment]
Hi! Nice to meet everyone. I'm currently working on this and will be able to provide an update on the status and an ETA later in the week.

**Comment by fxgamundi** [comment]
Hey! Progress is going well and we should have a it ready for review by the end of next week.

**Comment by Noc2** [comment]
That’s great to hear.

**Comment by fxgamundi** [comment]
We're doing a final internal review before submitting the updated code. We'll follow with the PR later today

**Comment by Noc2** [comment]
Sounds good. It’s not super urgent. I’m currently sick (covid) and can not work much anyway ;-)

**Comment by fxgamundi** [comment]
Sorry to hear that, get better!  In that case I will take some additional time to clean a couple of things.

**Comment by fxgamundi** [comment]
Hi! The PR for the parachain staking page is up! https://github.com/polkadot-js/apps/pull/7275

**Review by Noc2 — APPROVED** [review]
Sorry for the late reply here. I’m happy to tell you that the milestone is a pass and thanks for all the extra effort you put into this. I will forward your invoice internally.

**Comment by meldien** [comment]
Hi all, is it possible to update the invoice date?

**Comment by 4meta5** [comment]
@meldien the invoice was submitted via the google form from what I remember? would you like us to submit a new invoice via the google form? Feel free to send the old invoice to `stefan@purestake.com` and we can also figure this out over email

**Comment by meldien** [comment]
@4meta5  @Noc2 yes that would be great if you can submit a new invoice during google form. because the date is from 2021 and the approval are done in 2022.

**Comment by meldien** [comment]
@4meta5 any update?

**Comment by 4meta5** [comment]
@meldien we have assigned the task of sending the new invoice to someone else internally and will submit it soon @crystalin

**Comment by fededubbi** [comment]
Hi all, any updates here?

Many thanks,
Federica

**Comment by 4meta5** [comment]
I submitted an updated invoice. Let me know if there are any problems with the submission.

**Comment by RouvenP** [comment]
hi @4meta5 we transferred the payment today. Thanks!

**Pull request merged**